OsmAnd/src/net/osmand/plus/settings/vehiclesize/FuelCapacityHelper.java

```diff
@@ -0,0 +1,99 @@
+package net.osmand.plus.settings.vehiclesize;
+
+import androidx.annotation.NonNull;
+
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.R;
+import net.osmand.plus.settings.backend.ApplicationMode;
+import net.osmand.plus.settings.backend.OsmandSettings;
+import net.osmand.plus.settings.backend.preferences.OsmandPreference;
+import net.osmand.plus.settings.enums.VolumeUnit;
+import net.osmand.plus.utils.OsmAndFormatter.FormattedValue;
+import net.osmand.plus.widgets.chips.ChipItem;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class FuelCapacityHelper {
+
+	public static final float IMPERIAL_GALLONS_IN_LITER = 4.546091879f;
+	public static final float US_GALLONS_IN_LITER = 3.785411784f;
+
+	public static FormattedValue getFormattedValue(@NonNull OsmandApplication app, @NonNull VolumeUnit volumeUnit, @NonNull ApplicationMode mode) {
+		OsmandSettings settings = app.getSettings();
+		boolean separateWithSpace = false;
+		String textValue;
+		String textUnit;
+		float value = readSavedValue(settings, volumeUnit, mode);
+		if (value == 0.0f) {
+			textValue = app.getString(R.string.shared_string_none);
+			textUnit = "";
+		} else {
+			DecimalFormat formatter = new DecimalFormat("0.#", new DecimalFormatSymbols(Locale.US));
+			textValue = formatter.format(value);
+			textUnit = volumeUnit.getUnitSymbol(app);
+			separateWithSpace = true;
+		}
+		return new FormattedValue(value, textValue, textUnit, separateWithSpace);
+	}
+
+	public static float readSavedValue(@NonNull OsmandSettings settings, @NonNull VolumeUnit volumeUnit, @NonNull ApplicationMode mode) {
```

Comment:
It will be better to remove FuelCapacityHelper, use OsmAndFormatter for UI & store always in litres